### DISCUSSION TITLE
Watchtower seems to be dead and is no longer being developed?

### DISCUSSION THREAD

**Discussion by NetRat88** [containrrr/watchtower#1993] [General]
Hey together,

is watchtower still under development?
The last commits are more than 8 months ago and it seems that it will not be developed anymore isn't?

Is there any alternative or is there hope that things will continue here?

Thanks a lot!

**Comment by Criomby** [comment]
I really hope this project will be actively developed in the future but the lack of activity starts showing as issues and pull requests are stacking up without any word of the repo owner. Not even an absence notice or something.

In the meantime I looked for alternatives as I was having some issues with watchtower and found these projects:

- [diun](https://github.com/crazy-max/diun) (seems to be the most watchtower-like alternative)
- [whats-up-docker](https://github.com/fmartinou/whats-up-docker) (includes a web UI, kind of like [dockge](https://github.com/louislam/dockge))
- [dockcheck](https://github.com/mag37/dockcheck) (offers a simple CLI which I really like)

**Reply by huyz** [reply]
What kind of issues have you been running into? I've been using Watchtower for a month and I'm wondering what I need to watch out for

**Reply by Criomby** [reply]
Since I'm using watchtower to watch (!) my containers only as I don't want critical applications to be updated with potentially breaking changes automatically while I'm away, my experience may be different from yours.

Anyways, I am having issues with the reliability of pull actions as watchtower started having issues with some registries while they still work just fine from the command line and those errors pass silently even with email notifications enabled. The other thing is that watchtower messes with my images and removes tags (e.g. `:latest` gets changed to `:<none>`) after pre-pulling and checking for changes, which leads to issues when rebuilding containers or with other tools that rely on that information.

**Comment by NetRat88** [comment]
Someone seems to fork this repo and will move forward 🥳

https://github.com/nicholas-fedor/watchtower

**Reply by waja** [reply]
I'm using https://github.com/nicholas-fedor/watchtower since a couple of weeks on more then a dozen of systems and it works very well so far.

**Reply by huyz** [reply]
Any notable improvements?

**Reply by NetRat88** [reply]
You can check the release notes

https://github.com/nicholas-fedor/watchtower/releases

**Reply by huyz** [reply]
I skimmed them and all this is too technical for me.  I'm wondering if there are some notable _and_ easy-to-understand improvements

**Reply by bondskin** [reply]
@huyz i switched from the original docker to this one. Works like a charm, no issues

There are no new features as much as I can tell. It does it's work and it is maintained.

**Reply by nicholas-fedor** [reply]
> I skimmed them and all this is too technical for me. I'm wondering if there are some notable _and_ easy-to-understand improvements

Some of the more substantial updates:

- Updates to better maintain networking configurations, including static MAC addressing.
- Updated support for container registries, such as GHCR, when requesting container image information.
- The API can now be used to specify image tags for updates (thanks to @Foxite).

This omits the numerous updates that improve future maintainability, resolve various bugs, performance improvements, logging standardization, etc.

It's also worthwhile to mention that I also maintain a fork of Shoutrrr, which provides notification functionality to Watchtower. This has allowed me to provide updates there, such as updating Teams, adding Discord threads, and adding Lark support, amongst a variety of similar maintenance/debugging-related improvements.

**Reply by huyz** [reply]
@nicholas-fedor Thanks so much for your forks

**Reply by LayerCakeMakes** [reply]
It would be great if @simskij could pass on the project to @nicholas-fedor as this project just seems dead otherwise.
